addons/procurement/procurement.py

```diff
@@ -253,51 +215,45 @@ def _assign(self, cr, uid, procurement, context=None):
             :rtype: boolean
         '''
         #if the procurement already has a rule assigned, we keep it (it has a higher priority as it may have been chosen manually)
-        if procurement.rule_id:
+        if self.rule_id:
```

Comment:
Use self.ensure_one() to verify that the self contains only one record or make the method to work on multiple records.